### PR TITLE
Add 3-player (sanma) mahjong support + minor bugfix

### PR DESCRIPTION
**Pull request by smly**
Resolves https://github.com/smly/RiichiEnv/issues/30

## Summary

Add support for 3-player mahjong (sanma / 三麻) to RiichiEnv, including game state management, observation encoding, hand evaluation, yaku detection, scoring, and Python bindings.

### Key changes

- **Game state (`state_3p/`)**: New `GameState3P` with 108-tile wall (no 2m–8m), 3 players, and sanma-specific rules (e.g., tsumo-loss scoring, North wind kita/nukidora handling)
- **Kita (抜きドラ) action**: Implement BaBei (北抜き) action type, allowing players to declare North tiles as bonus dora, with rinshan draw and chankan-ron checks
- **Hand evaluator (`hand_evaluator_3p.rs`)**: Dedicated evaluator for the reduced tile set (27 tile types)
- **Yaku detection (`yaku_3p.rs`)**: 3-player yaku evaluation with nukidora counting and adjusted yaku conditions
- **Observation encoding (`observation_3p/`)**: Compact observation space for 3-player games, including kita-related feature planes
- **Action encoding**: Compact 60-action space for 3-player games (vs. 83 for 4-player) via `ActionEncoder::ThreePlayer`, mapping only 1m/9m + full pin/sou/honors
- **Game variant abstraction (`game_variant.rs`)**: `GameStateVariant` enum to dispatch between `GameState` (4P) and `GameState3P` (3P) transparently
- **Python bindings**: `RiichiEnv` now supports `GameType.SAN_HANCHAN` / `SAN_TONPUU`, returning `Observation3P` objects; dispatch via `with_variant!` / `with_variant_mut!` macros
- **Scoring (`score.rs`, `rule.rs`)**: Tsumo-loss (ツモ損) and per-player point distribution for 3-player games
- **Replay parsing (`replay/`)**: MjSoul 3-player replay ingestion support
- **Tests (`tests/env/test_sanma.py`, `tests.rs`)**: Comprehensive test suite covering initialization, kita handling, game progression, scoring, and edge cases

### CODE REVIEW

File: riichienv-core/src/score.rs [hunk 1]
```diff
@@ -19,37 +27,45 @@ pub fn calculate_score(han: u8, fu: u8, is_oya: bool, is_tsumo: bool, honba: u32
...
-    calculate_score(han, fu, is_oya, is_tsumo, honba)
+pub fn calculate_score_py(
+    han: u8,
+    fu: u8,
+    is_oya: bool,
+    is_tsumo: bool,
+    honba: u32,
+    num_players: Option<u8>,
+) -> Score {
+    calculate_score(han, fu, is_oya, is_tsumo, honba, num_players.unwrap_or(4))
```
[the remark this answers is not in the document]
Comment (smly) [reply riichienv-core/src/score.rs]: Good catch — the `calculate_score_py` in `score.rs` was dead code (compiled with `#[cfg(feature = "python")]` but never registered in the `#[pymodule]`). Removed it entirely.

The actual Python binding in `riichienv-python/src/lib.rs` uses `#[pyo3(signature = (han, fu, is_oya, is_tsumo, honba, num_players=4))]`, which makes `num_players` optional with a default of 4 at the Python level while passing a concrete `u8` to the Rust core function. This preserves backward compatibility — existing 4-player code like `calculate_score(4, 30, False, True, 0)` continues to work without modification, while 3-player callers can explicitly pass `num_players=3`.
